### PR TITLE
Add paragraph about usage of parent-child relationship

### LINKED ISSUE #1282: Messaging: explain that parent could be used in addition to link
Extracted from https://github.com/open-telemetry/semantic-conventions/issues/1205

- add spec text allowing consumer span to parent to message
- add parenting to examples
- maybe explain why link as necessary

### PR DESCRIPTION
**Pull request by joaopgrassi**
Fixes #1282

## Changes

Adds a paragraph mentioning the possibility to use the message creation context as a parent of "Process" spans, to achieve a more traditional parent-child trace structure.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

File: docs/messaging/messaging-spans.md [hunk 1]
```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
...
 "Receive" span SHOULD link to the message's creation context.
 
+> [!IMPORTANT]
+> These conventions use spans links as the default mechanism to correlate
+> producers and consumer(s) because:
+>
+> - It is the only consistent trace structure that can be guaranteed,
+> given the many different messaging systems models available.
+>
+> - It is the only option to correlate produce and consumer(s) in batch scenarios
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:271]: nit

```suggestion
> - It is the only option to correlate producer and consumer(s) in batch scenarios
```

File: docs/messaging/messaging-spans.md [hunk 2]
```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
...
+> as a span can only have a single parent.
+>
+> - Offers a consistent experience for users analysing traces from different
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:274]: It seems very similar to the first point, no?

> - It is the only consistent trace structure that can be guaranteed, given the many different messaging systems models available.

File: docs/messaging/messaging-spans.md [hunk 3]
```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
...
+> messaging systems.
+
+Exclusively for single messages scenarios, "Process" or "Receive" span MAY
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:277]:
```suggestion
Exclusively for single messages scenarios, "Process" span MAY
```

Receive spans can't use message context as a parent because they start before message is received and the context is known.

File: docs/messaging/messaging-spans.md [hunk 4]
```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
...
+>
+> - It is the only consistent trace structure that can be guaranteed,
+> given the many different messaging systems models available.
+>
+> - It is the only option to correlate produce and consumer(s) in batch scenarios
+> as a span can only have a single parent.
+>
+> - Offers a consistent experience for users analysing traces from different
+> messaging systems.
+
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:276]:
```suggestion
> - It is the only option to correlate produce and consumer(s) when message consumption can happen in the scope of another ambient context such as HTTP server span.
```

File: docs/messaging/messaging-spans.md [hunk 5]
```diff
@@ -261,6 +261,23 @@ batch of messages, or for no message at all (if it is signalled that no
...
+Exclusively for single messages scenarios, "Process" or "Receive" span MAY
+use the message's creation context as its parent, thus achieving a direct
+parent-child relationship between producer and consumer(s).
```
Comment (lmolkova) [inline docs/messaging/messaging-spans.md:279]: I'm suggesting to add a bit of additional guidance around the ambient context and defaults.

```suggestion
parent-child relationship between producer and consumer(s). Instrumentations SHOULD document whether they use message creation context as a parent for "Process" spans and MAY provide configuration option allowing users to control this behavior. 

It's NOT RECOMMENDED to use message creation context as a parent for "Process" span (by default) if processing usually happens in the scope of another span.

If instrumentation uses message creation context as a parent on "Process" span in the scope of another valid ambient context, it SHOULD set the ambient context as a link on "Process" span to preserve correlation between message processing and that context. 

For example, a messaging broker pushes messages to consumers over HTTP and a consumer application has HTTP server and messaging instrumentations enabled. The messaging instrumentation would create "Process" span following one of these possible approaches:

- "Process" span is a child of HTTP server span context and has a link to message creation context. This is the default behavior for this instrumentation.
- "Process" span is a child of message creation context and has two links: one to message creation context and another one to HTTP server span context. This is opt-in behavior.
```

LMK what you think. 

Maybe we can also put it all into a new section like "Using message context as a parent for "Process" spans" so it appears in the TOC?

**Review by lmolkova — APPROVED** [review]
LGTM, suggesting to add a bit more guidance for parent case

File: docs/messaging/messaging-spans.md (hunk 3 above)
Comment (joaopgrassi) [reply docs/messaging/messaging-spans.md:277]: Ah yeah of course!

File: docs/messaging/messaging-spans.md (hunk 2 above)
Comment (joaopgrassi) [reply docs/messaging/messaging-spans.md:274]: Yeah good point. I tried to merge them together but didn't sound good. I will just remove it then :)

File: docs/messaging/messaging-spans.md (hunk 5 above)
Comment (joaopgrassi) [reply docs/messaging/messaging-spans.md:279]: I was thinking the same, but to add new headers for `Receive` spans and `Process` inside the `Consumer spans` section. Right now they are all together and I find it a bit hard to follow.

Comment (lmolkova) [reply docs/messaging/messaging-spans.md]: Thank you! I like "Message creation context as parent of "Process" span"!

**Review by lmolkova — APPROVED** [review]
